server/xpub/entities/manuscript/resolvers.test.js

```diff
@@ -28,10 +32,12 @@ describe('Submission', () => {
     const user = await User.save(userData)
     userId = user.id
     mailer.clearMails()
+    initializing = false
```

Comment:
We shouldn't need to have an `initializing` parameter. Jest will wait for the async function to complete before running the tests.

Reply:
You can see details [here](https://jest-bot.github.io/jest/docs/setup-teardown.html) and [here](https://github.com/facebook/jest/issues/1256)

Reply:
You're right we shouldn't. However I'd like to leave them in - that way if anyone introduces code that makes the tests fail because of async init or tear down then we will easily know.

Reply:
It may be worth using a promise to make it clearer that we're waiting for the database to be initialised. Using an assertion says to me that we're testing the database initialisation.

i.e.
```
 let initializeDb;

 const initialize = async () => {
    replaySetup('success')
    await Promise.all([
      fs.remove(config.get('pubsweet-server.uploads')),
       createTables(true),
     ])
    const user = await User.save(userData)
    userId = user.id
    mailer.clearMails()
 }

 beforeEach(async () => {
    initializeDb = initialize();
 });

 ...

 it('Returns empty object when there are no manuscripts in the db', async () => {
      await initializeDb();
      const manuscript = await Query.currentSubmission({}, {}, { user: userId })
      expect(manuscript).toBe(null)
 })
```

Reply:
Ooo yeah I like that :+1: 
.. you want to add it? @damnhipster

Reply:
This `initializing` stuff is unnecessary noise. All it is asserting is that the `beforeEach` function completed before the test started but this is guaranteed by Jest already. 

If a bug in the initialisation logic is introduced (e.g. forgetting to await a statement) this won't catch it because the `beforeEach` still completed and set `initializing` back to `false`. The most likely way that assertion could fail is if an exception is thrown in the `beforeEach`. But in that case it's confusing because the initialisation failed so `initializing` should really be `false`.

The promise solution is bypassing Jest's automatic waiting in favour of explicit manual waiting. Surely that's likely to lead to more bugs, not fewer.

Reply:
I'd like to leave it in there... it helped me find the issue in pubsweet with the asynchronous drop tables. So it would catch issues with pubsweet? Please correct if I am wrong/you disagree.